code/modules/mob/living/death.dm

```diff
@@ -74,6 +74,17 @@
 
 	if (client)
 		client.move_delay = initial(client.move_delay)
+		if(!suiciding && !SSticker.first_death.len) //suicides don't clog up the list, only BRUTAL deaths here
+			var/list/L = list()
+			var/area/A = get_area(src)
+			L["name"] = name
+			L["role"] = null
+			if(mind.assigned_role)
+				L["role"] = mind.assigned_role
+			L["area"] = A.map_name

```

Comment:
`L["area"] = get_area_name(src, TRUE)`

Reply:
Would be sweet if you added the coordinates too

Reply:
Woah, that proc exists? Sweet.